src/doc/reference.md

```diff
@@ -1983,6 +1983,7 @@ macro scope.
 
 ### Miscellaneous attributes
 
+- `deprecated` - Allows using the `#[deprecated]` attribute.
```

Comment:
The syntax is `#[deprecated(since = "crate version", note = "...")]`, both fields optional. Documented [in the RFC](https://github.com/rust-lang/rfcs/blob/master/text/1270-deprecation.md) and [the test suite](https://github.com/rust-lang/rust/pull/30206/files#diff-95c6b15069c0b4cfc829a6ba2747cb35R13) but it really needs to be in the reference.